src/scratch/ScratchSound.as

```diff
@@ -81,40 +92,40 @@ public class ScratchSound {
 				sampleDataStart = info.sampleDataStart;
 				reduceSizeIfNeeded(info.channels);
 			} catch (e:*) {
+				var extraData:Object = {
+					exception: e, info: info
+				};
+				if (e is Error) {
+					var error:Error = e as Error;
+					Scratch.app.log(LogLevel.WARNING, 'Error while constructing sound:' + e.message, extraData);
+				}
+				else {
+					Scratch.app.log(LogLevel.WARNING, 'Unknown error while constructing sound', extraData);
+				}
 				setSamples(new Vector.<int>(0), 22050);
 			}
 		}
 	}
 
-	public function get soundData():ByteArray {
-		return __soundData;
-	}
-
-	public function set soundData(data:ByteArray):void {
-		__soundData = data;
-		md5 = null;
-	}
-
 	private function reduceSizeIfNeeded(channels:int):void {
 		// Convert stereo to mono, downsample if rate > 32000, or both.
 		// Compress if data is over threshold and not already compressed.
 		const compressionThreshold:int = 30 * 44100; // about 30 seconds
 		if (rate > 32000 || channels == 2 || format == 'float') {
 			var newRate:int = (rate > 32000) ? rate / 2 : rate;
 			var samples:Vector.<int> = WAVFile.extractSamples(soundData);
-			if (rate > 32000 || channels == 2)
-				samples = (channels == 2) ?
-					stereoToMono(samples, (newRate < rate)) :
-					downsample(samples);
+			if (rate > 32000 || channels == 2) {
+				samples = (channels == 2) ? stereoToMono(samples, (newRate < rate)) : downsample(samples);
+			}
 			setSamples(samples, newRate, true);
-			soundID = 0;
```

Comment:
Sound IDs are array indices so zero is a valid sound ID and could refer to some other sound. My guess is that the intent was to invalidate the sound ID, but `setSamples` already does that so I believe it's safe to remove this line.